packages/mui-joy/src/Input/Input.tsx

```diff
@@ -334,12 +334,14 @@ const Input = React.forwardRef(function Input(inProps, ref) {
 
   return (
     <SlotRoot {...rootProps}>
-      {startDecorator && (
+      {(startDecorator ?? slots?.startDecorator) && (
```

Comment:
I think we should not do this. The intention of `slots.*` is to replace the default slot component, there shouldn't be other logic involved.

Reply:
Okay, should we add something in docs explaining same? I think it's confusing right now

Reply:
Agree!

Reply:
We already have the overall doc about `slots` at https://mui.com/joy-ui/customization/overriding-component-structure/.